apps/frontend/app/(client)/(main)/settings/page.tsx

```diff
@@ -198,133 +170,102 @@ export default function Page() {
     }
   }
 
-  const onSubmitClick = () => {
-    // submit 되기위해, watch로 확인되는 값이 default값과 같으면 setValue를 통해서 defaultProfileValues로 변경
-    if (realName === '') {
-      setValue('realName', defaultProfileValues.userProfile?.realName)
-    }
-    if (majorValue === defaultProfileValues.major) {
-      setMajorValue(defaultProfileValues.major)
-    }
-    if (currentPassword === '') {
-      setValue('currentPassword', 'tmppassword1')
-    }
-    if (newPassword === '') {
-      setValue('newPassword', 'tmppassword1')
-    }
-    if (confirmPassword === '') {
-      setValue('confirmPassword', 'tmppassword1')
-    }
+  const resetToSubmittableValue = (
+    field: 'realName' | 'currentPassword' | 'newPassword' | 'confirmPassword',
+    value: string | undefined,
+    defaultValue: string
+  ) => {
+    if (value === '') setValue(field, defaultValue)
   }
 
-  const checkPassword = async () => {
-    setIsCheckButtonClicked(true)
-    try {
-      const response = await safeFetcher.post('auth/login', {
-        json: {
-          username: defaultProfileValues.username,
-          password: currentPassword
-        }
-      })
+  const onSubmitClick = () => {
+    resetToSubmittableValue(
+      'realName',
+      realName,
+      defaultProfileValues.userProfile?.realName ?? ''
+    )
+    resetToSubmittableValue('currentPassword', currentPassword, 'tmppassword1')
+    resetToSubmittableValue('newPassword', newPassword, 'tmppassword1')
+    resetToSubmittableValue('confirmPassword', confirmPassword, 'tmppassword1')
+  }
 
-      if (response.status === 201) {
-        setIsPasswordCorrect(true)
-        setNewPasswordAble(true)
-      }
-    } catch {
-      console.error('Failed to check password')
-    }
+  const settingsContextValue: SettingsContextType = {
```

Comment:
```suggestion
  const settingsContextValue = {
```

타입 추론이 되기 때문에 타입 지정은 필요 없을 것 같습니다!